### PR TITLE
ocamlPackages.mm: init at 0.8.1

### PR DESCRIPTION
**Pull request by D4ndellion**
###### Description of changes
Part of splitting out https://github.com/NixOS/nixpkgs/pull/140777 for easier review

also packages its dependencies

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/ao/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ lib, buildDunePackage, fetchFromGitHub, dune-configurator, libao }:
+
+buildDunePackage rec {
+  pname = "ao";
+  version = "0.2.4";
+
+  minimalOCamlVersion = "2.0";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/ao/default.nix:7]: This line is surprising…

File: pkgs/development/ocaml-modules/mm/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  src = fetchFromGitHub {
+    owner = "savonet";
+    repo = "ocaml-mm";
+    rev = "v${version}";
+    sha256 = "sha256-7ozt+OgKNxMnjl2R+/ce27ZyL+T6BShvnnFE5BasJC4=";
+  };
+
+  buildInputs = [ dune-configurator ];
+  propagatedBuildInputs = [ alsa ao mad pulseaudio theora ]; # ocamlsdl is blocked in nixpkgs from building for ocaml >= 4.06
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/mm/default.nix:19]: I don’t understand the comment. What does it mean?

Comment (dali99) [reply pkgs/development/ocaml-modules/mm/default.nix:19]: ocamlsdl is an optional dependency for mm, but the minimalOcamlversion in mm is above the maximum version allowed for  ocamlsdl in nixpkgs. See https://github.com/NixOS/nixpkgs/blob/a43da8cc8b1f2ed2849f14b8f8f18f813883b7e3/pkgs/development/ocaml-modules/ocamlsdl/default.nix#L7-L9

File: pkgs/development/ocaml-modules/ao/default.nix (hunk 1 above)
Comment (dali99) [reply pkgs/development/ocaml-modules/ao/default.nix]: I agree, can't figure out why I marked it as such I might have misread read dune >= 2.0 as ocaml >= 2.0

**Pull request merged**